posts/marker-gene-smoothing/index.qmd

```diff
@@ -0,0 +1,627 @@
+                             ,xlim=NULL,ylim=NULL){
+  
+  pd <-  
+    data.table(sem@reductions[[umapreduc]]@cell.embeddings
+               ,keep.rownames = TRUE)
+  setnames(pd, c(names(pd)[1:3]), c("cell_ID", "UMAP_1", "UMAP_2"))
+  pd[match(colnames(marker_matrix), cell_ID),(c(marker)):=marker_matrix[marker,]]
+  setkeyv(pd, marker)
+  if(scale_expr){
+    p <- 
+      ggplot(pd 
+             ,aes(x=UMAP_1, y = UMAP_2, color=scale(.data[[marker]]))) 
+      
+  } else {
+    p <- 
+      ggplot(pd 
+             ,aes(x=UMAP_1, y = UMAP_2, color=.data[[marker]])) 
+    
+  }
+  p <- p +
+    geom_point(size=0.1) +
+    scale_color_gradientn(colors=brewer.pal(9, "Reds")) + 
+    theme_bw() +
+    scale_x_continuous(n.breaks=10) + 
+    scale_y_continuous(n.breaks=10) + 
+    labs(title=marker) + coord_fixed(xlim=xlim,ylim=ylim) 
+  return(p)
+}
+
+```
+
+```{r}
+#| eval: false
+#| message: false
+#| code-fold: true
+
+raw_foxp3_plot <- 
+  marker_umap_plot(sem[["RNA"]]@counts, "FOXP3", sem, "pearsonumap"
+                   ,xlim = c(-5.5, 6.5), ylim = c(-0.01, 7.2),scale=FALSE) + 
+    labs(title="raw FOXP3")
+
+ctpal_sub <- ctpal
+ctpal_sub[grep("^T",names(ctpal_sub),invert=TRUE)] <- "grey77"
+cowplot::plot_grid(
+  raw_foxp3_plot
+  ,umapf("pearsonumap", "clust", sem
+         ,cls = ctpal_sub, xlim = c(-5.5, 6.5), ylim = c(-0.01, 7.2))
+  ,nrow=2
+  ) + 
+  theme(plot.background = element_rect(fill = 'white',color='white')) + 
+  cowplot::panel_border(remove=TRUE) 
+
+```
+
+```{r}
+#| eval: true
+#| echo: false
+
+knitr::include_graphics("./figures/foxp3raw.png")
+```
+
+# Using 'smoothed' FOXP3 to highlight Treg focal point and refine cell types.
+
+One potential approach to cleaning this up is to use some form of 'smoothed' marker gene expression, rather than the raw FOXP3 counts.
+
+We can describe the calculation for 'smoothed' marker expression in two basic steps:
+
+1. Find the $K$ nearest neighbors of each cell in UMAP space.
+2. Average the raw (or normalized) expression of the gene across that cells nearest neighbors.
+
+There's a few different ways we might think about motivating this approach.  The UMAP is an approximate manifold projection, which takes a low (2-d) representation of our high-dimensional (960 genes) data set.
+The manifold is 'locally connected', meaning cells near each other in UMAP space are also similar to each other in expression space.
+By averaging the expression of a marker gene of interest across cells with similar profiles, we can make a simple imputation of what we'd expect to see for the marker gene, given other cells of similar expression profiles.
+
+
+This function below identifies the nearest neighbors in *UMAP* space, and makes a smoothing matrix, which can be used to get the average expression of any gene among a given cell's nearest neighbors.
+
+```{r,eval=FALSE,message=FALSE}
+
+## create a column-standardized nearest neighbor matrix 
```

Comment:
Also, if convenient, could you reformat it to use a counts matrix and a umap projection instead of a seurat object? Though I acknowledge my own bias here; happy for you to keep as-is if you think it's the right way to go.

Reply:
Probably worth a few comments on what's happening within the major blocks of this function.